torchdata/dataloader2/reading_service.py

```diff
@@ -247,13 +247,13 @@ def initialize(self, datapipe: DataPipe) -> DataPipe:
                 ctx, round_robin_dps, process_name="dispatching process"
             )
             assert len(req_queues) == self.num_workers and len(res_queues) == self.num_workers
-            process.daemon = True
-            process.start()
-            self._dispatch_process = (process, req_queues, res_queues)
             for req_queue in req_queues:
                 req_queue.cancel_join_thread()
             for res_queue in res_queues:
                 res_queue.cancel_join_thread()
+            process.daemon = True
+            process.start()
+            self._dispatch_process = (process, req_queues, res_queues)
```

Comment:
non-blocking note: Maybe we should refactor out the dispatching code at some point... It is getting too much

Reply:
Yeah. We should spin off it and worker launch to a separate file